reference_validator/YAML_HotValidator.py

```diff
@@ -0,0 +1,724 @@
+#!/usr/bin/env python
+#coding=utf-8
+
+# File: YAML_HotValidator.py
+# Brief: Contains class YAML_HotValidator for validating references
+# Author: Katerina Pilatova (kpilatov)
+# Date: 2016
+
+from __future__ import with_statement, print_function
+
+import os
+import pprint
+import sys
+import six  # compatibility
+import yaml # pip install pyyaml
+
+import YAML_Enums as ENUM
+import YAML_Hotfile as HOT
+import YAML_HotClasses
+
+# only for Python 2.x
+if sys.version_info[0] == 2:
+    import nyanbar
+
+class YAML_HotValidator:
+    ''' Detects unused variables, invalid references.'''
+
+    def __init__(self, arguments):
+        ''' Finds *.yaml files based on entered arguments.
+            arguments - dictionary with parsed arguments and their values
+        '''
+
+        # in environments, mappings, templates: all nodes with references to parent/children
+        # in curr_nodes: currently validated nodes (DFS - depth-first search)
+
+        # Save initial directory
+        self.init_dir = os.getcwd()
+
+        # List of YAML files to be checked + referenced children nodes
+        self.environments = []
+        self.mappings = []
+        self.templates = []
+        self.parameters = {}
+
+        # Currently opened nodes
+        self.curr_nodes = []
+
+        # Applied parameters
+        self.print_unused = arguments['print_unused']
+        self.pretty_format = arguments['pretty_format']
+        self.print_structure = arguments['print_tree']
+        self.print_nyan = (sys.version_info[0] == 2) and arguments['nyan']
```

Comment:
With the conditional import above, this can now be just:

    self.print_nyan = nyanbar and arguments['nyan']